src/gtpu/encap.c

```diff
@@ -835,18 +852,22 @@ static int gtp5g_fwd_skb_encap(struct sk_buff *skb, struct net_device *dev,
 
     pdr->ul_pkt_cnt++;
     pdr->ul_byte_cnt += skb->len; /* length without GTP header */
-    GTP5G_INF(NULL, "PDR (%u) UL_PKT_CNT (%llu) UL_BYTE_CNT (%llu)", pdr->id, pdr->ul_pkt_cnt, pdr->ul_byte_cnt);
-
+    GTP5G_INF(NULL, "PDR (%u) UL_PKT_CNT (%llu) UL_BYTE_CNT (%llu)", pdr->id, pdr->ul_pkt_cnt, pdr->ul_byte_cnt);    
+ 
+    if (pdr->urr_num != 0) {
+        if (update_urr_counter_and_send_report(pdr, far, volume, volume_mbqe, true) < 0)
+            GTP5G_ERR(pdr->dev, "Fail to send Usage Report");
+    }
+    
+    if (color == Red) {
+        GTP5G_TRC(pdr->dev, "Drop red packet");
+        return PKT_DROPPED;
+    }
     ret = netif_rx(skb);
     if (ret != NET_RX_SUCCESS) {
         GTP5G_ERR(dev, "Uplink: Packet got dropped\n");
```

Comment:
Do we need to free the skb when netif_rx fails?

Reply:
”netif_rx“ will free the skb when netif_rx fails